src/Sparrow/Json/BlittableJsonReaderVector.cs

```diff
@@ -0,0 +1,568 @@
+        }
+        
+        public bool IsOfType<T>()
+        {
+            var type = typeof(T);
+            switch (_header->Type)
+            {
+                case BlittableVectorType.SByte:
+                    return type == typeof(sbyte);
+                case BlittableVectorType.Int16:
+                    return type == typeof(short);
+                case BlittableVectorType.Int32:
+                    return type == typeof(int);
+                case BlittableVectorType.Int64:
+                    return type == typeof(long);
+                case BlittableVectorType.Byte:
+                    return type == typeof(byte);
+                case BlittableVectorType.UInt16:
+                    return type == typeof(ushort);
+                case BlittableVectorType.UInt32:
+                    return type == typeof(uint);
+                case BlittableVectorType.UInt64:
+                    return type == typeof(ulong);
+#if NET6_0_OR_GREATER
+                case BlittableVectorType.Half:
+                    return type == typeof(Half);
+#endif
+                case BlittableVectorType.Float:
+                    return type == typeof(float);
+                case BlittableVectorType.Double:
+                    return type == typeof(double);
+                default:
+                    return false;
+            }
+        }
+
+        /// <summary>
+        /// Useful when we've to copy memory without known type. 
+        /// </summary>
+        /// <returns>Span of underlying memory</returns>
+        public ReadOnlySpan<byte> ReadUnderlyingMemory()
+        {
+            return new ReadOnlySpan<byte>(_dataStart, _header->Count * _header->ElementSize);
+        }
+
+        public ReadOnlySpan<T> ReadArray<T>()
+            where T : unmanaged
+        {
+            if (IsOfType<T>() == false)
+                throw new InvalidOperationException($"Vector is not of type {typeof(T).Name}");
+
+            return new ReadOnlySpan<T>(_dataStart, _header->Count);
+        }
+
+        public bool TryReadArray<T>(out ReadOnlySpan<T> span)
+        {
+            if (IsTypeCompatibleForDirectRead<T>() == false)
+            {
+                span = [];
+                return false;
+            }
+
+            span = new ReadOnlySpan<T>(_dataStart, _header->Count);
+            return true;
+        }
+
+        public EnumerableAs<TOut> ReadAs<TOut>() where TOut : unmanaged
+        {
+            var size = _header->Count * _header->ElementSize; 
+            return new EnumerableAs<TOut>(_dataStart, size, ElementSize, _header->Type);
+        }
+
+        /**
+         * Since compression may happen on the server side, we need a simple way to deserialize it on the client side.
+         * Let's introduce a method that reads the requested type.
+         **/
+        public struct EnumerableAs<TOut> : IEnumerator<TOut>
+            where TOut : unmanaged
+        {
+            private readonly BlittableVectorType _type;
```

Comment:
Please check below warning `The field 'BlittableJsonReaderVector.EnumerableAs<TOut>._type' is never used`